### PR TITLE
Add explicit stderr-vs-ierr contract coverage

### LINKED ISSUE #35: Add explicit stderr-vs-ierr contract coverage
# Add explicit stderr-vs-`ierr` contract coverage

## Why
Deep review issue [#29](https://github.com/jaharris87/fTimer/issues/29) found that the documented error contract is implemented, but not actually defended by tests that verify stderr silence vs warning behavior.

Affected areas:
- `src/ftimer_core.F90`
- `src/ftimer_core_summary_bindings.F90`
- `tests/test_edge_cases.pf`
- `tests/test_nesting.pf`

## Scope
- Add tests that exercise representative error paths with `ierr` present and absent.
- Capture or otherwise verify `error_unit` behavior for those cases.
- Cover core runtime and summary/report entry points where practical.

## Acceptance Criteria
- Tests explicitly verify silence when `ierr` is present.
- Tests explicitly verify warning output when `ierr` is absent.
- The behavior is covered for more than one code path, not just a single edge case.

## Deep Review Links
- Umbrella: [#29](https://github.com/jaharris87/fTimer/issues/29)
- Review roles: Methodology Review, Test Quality Review

### PR DESCRIPTION
**Pull request by jaharris87**
Closes #35

## Summary
- add explicit stderr capture helpers for pFUnit contract checks
- verify warn-mode mismatch handling stays silent when `ierr` is present and emits a warning when `ierr` is omitted
- verify `print_summary` before-init handling stays silent when `ierr` is present and emits a warning when `ierr` is omitted

## Representative code paths covered
- Core runtime: warn-mode mismatch path in `ftimer_core` via `stop("A")` with `B` on top of the stack
- Summary/report entry point: `print_summary()` before init in `ftimer_core_summary_bindings`

## stderr verification strategy
- added a tiny test-only helper in `tests/test_support.F90` that temporarily redirects file descriptor 2 to a temp file, flushes `error_unit`, restores stderr, and reads the captured text back for assertions
- assertions check both silence and presence of stable key substrings so wording can evolve without making the tests brittle

## Implementation / docs
- no runtime behavior change was needed
- no doc changes were needed because the existing contract text already matches the implementation these tests now defend

## Tests added
- `test_print_summary_before_init_with_ierr_is_silent`
- `test_print_summary_before_init_without_ierr_warns_to_stderr`
- strengthened `test_warn_mismatch_repairs_and_returns_code` with stderr silence checks
- strengthened `test_warn_mismatch_without_ierr_repairs_and_continues` with stderr warning checks

## Verification
- `FC=gfortran cmake --fresh -B build -DFTIMER_BUILD_TESTS=ON -DPFUNIT_DIR=/Users/hrh/.local/PFUNIT-4.15/cmake`
- `cmake --build build`
- `ctest --test-dir build --output-on-failure`
- `cmake --fresh -B build-smoke`
- `cmake --build build-smoke`
- `ctest --test-dir build-smoke --output-on-failure`
- `find src -name '*.F90' -exec fprettify --diff {} +`
- `find tests -name '*.pf' -exec fprettify --diff {} +`
- `find tests -name '*.F90' -exec fprettify --diff {} +`
- `find examples -name '*.F90' -exec fprettify --diff {} +`

## Review workflow
- requested labels: `codex-software-review`, `codex-methodology-review`
- additional role requested: Test Quality Review

### CODE REVIEW

**Comment by jaharris87** [comment]
@codex review as a senior software engineer focused on correctness; find bugs, missing tests or docs, unsafe defaults, silent lossy behavior, and end-to-end validation gaps, not style issues; classify findings as bug, test gap, design concern, or nit; begin with "## Software Review".

File: tests/test_support.F90 [hunk 1]
```diff
@@ -43,6 +88,79 @@ subroutine attach_scripted_mock_clock(timer, values)
...
+      integer, intent(out) :: saved_fd
+      integer, intent(out) :: ierr
+      character(kind=c_char, len=:), allocatable :: c_path
+      integer(c_int) :: capture_fd
+      integer(c_int) :: saved_fd_c
+
+      call delete_file_if_exists(path)
+
+      c_path = trim(path)//c_null_char
+      saved_fd_c = c_dup(STDERR_FILENO)
```
[the remark this answers is not in the document]
Comment (jaharris87) [reply tests/test_support.F90]: Agree and fix: added a pre-redirect flush in begin_stderr_capture() (flush(error_unit) plus fflush(NULL)) before switching descriptor 2, so buffered stderr from earlier activity cannot leak into a later capture window. This was addressed in commit 50751be and reverified with cmake --build build and ctest --test-dir build --output-on-failure.
